sigchanyzer.go

```diff
@@ -48,17 +48,30 @@ func run(pass *analysis.Pass) (interface{}, error) {
 			return
 		}
 		var chanDecl *ast.CallExpr
+
+		// track whether call.Args[0] is a *ast.CallExpr. This is true when make(chan os.Signal) is passed directly to signal.Notify
+		inlinedChanMake := false
+
 		switch arg := call.Args[0].(type) {
 		case *ast.Ident:
 			if decl, ok := findDecl(arg).(*ast.CallExpr); ok {
 				chanDecl = decl
 			}
 		case *ast.CallExpr:
 			chanDecl = arg
+			inlinedChanMake = true
 		}
+
 		if chanDecl == nil || len(chanDecl.Args) != 1 {
 			return
 		}
+
+		// There is one exception. If we've got: signal.Notify(make(chan os.Signal), signal)
+		// the Go runtime will write to the channel with a non-blocking send. See https://golang.org/issues/45043
```

Comment:
This is not the main point. Whether you use `signal.Notify(make(chan os.Signal), signal)` or `signal.Notify(ch, signal)`, the runtime will always write to the channel using non-blocking send.

The former is safe, because nothing will read from the channel.